MusicNotationKit/MusicNotationKit/Measure.swift

```diff
@@ -46,12 +46,36 @@ public struct Measure: ImmutableMeasure, Equatable {
         noteCount = immutableMeasure.noteCount
         recomputeNoteCollectionIndexes()
     }
-
+	
+	public func note(at index: Int, inSet setIndex: Int) throws -> Note {
+		let noteCollectionIndex = try noteCollectionIndexFromNoteIndex(index, inSet: setIndex)
+		var noteIndex = 0
+		if noteCollectionIndex.tupletIndex != nil {
+			noteIndex = noteCollectionIndex.tupletIndex!
+		}
```

Comment:
You can rewrite this with the nil coalescing operator. You want to avoid `!` at all times. Always use optional unwrapping instead of this method. Here is how I would rewrite it:

`let noteIndex = noteCollectionIndex.tupletIndex ?? 0`